doc/read-the-docs-site/architecture/runtime-view.rst

```diff
@@ -0,0 +1,156 @@
+      Component(transformer, "Transformer", "Haskell component", "Applies some function to the incoming ProcessedInput and forwards them to the indexer")
+      Component(indexer, "Indexer", "Haskell thread", "Indexes the relevant parts of the ProcessedInput to some storage")
+      ContainerDb(marconiDb, "Marconi storage", "Storage backend of the indexer")
+    }
+
+    Rel(datasource, sourceReader, "sends some input data")
+    Rel(sourceReader, preprocessor, "sends '[ProcessedInput]'")
+    Rel(preprocessor, transformer, "sends preprocessed '[ProcessedInput]'")
+    Rel(transformer, indexer, "forwards each 'ProcessedInput'")
+    Rel(indexer, marconiDb, "stores events in the indexer storage")
+
+The input source reader provides an interface to fetch data from any datasource: a local Cardano node, Ogmios, cardano-db-sync, etc.
+However, the input data needs to be mapped to a ``ProcessedInput``.
+The ``ProcessedInput`` is sent to the preprocessor which applies some preprocessing function on the inputs.
+The preprocessed inputs are sent to the transformer which also transforms the inputs, and sends them to the indexer.
+Finally, the indexer will index them to it's storage backend.
+
+.. note::
+
+  See the explanation of the `marconi-core` components in order to understand concepts like ``Preprocessor``,  ``Transformer``, ``Indexer``, etc..
+
+As an example of an input source reader, here is a runtime view of a Cardano node source reader.
+
+.. uml::
+
+    !include <C4/C4_Dynamic>
+    title Cardano node source reader runtime view
+
+    ContainerDb(blockchain, "Blockchain")
+    Container_Ext(cardanoNode, "Local Cardano node")
+
+    Container_Boundary(marconiContainer, "Marconi application") {
+      Component(sourceReader, "Local node source reader", "Haskell component", "Module that processes data from the local node and transforms it to a [ProcessedInput]")
+    }
+
+    Rel(blockchain, cardanoNode, "sends blocks to")
+    Rel(cardanoNode, sourceReader, "sends chain-sync event")
+
+The local Cardano node constructs a local state of the Cardano blockchain by fetching blocks from its peers and storing it on disk.
+Then, node clients can request for blocks by the using the node's chain-sync mini-protocol.
+The chain-sync protocol essentially streams events to the client which have the following structure:
+
+.. code-block:: haskell
+
+  data ChainSyncEvent a
+    = RollForward a C.ChainTip
+    | RollBackward C.ChainPoint C.ChainTip
+    deriving (Show, Functor, Generic)
+
+where ``a`` can be anything including a block.
+
+Now, let's assume that we want to run multiple indexers that process inputs at the same rate, and that a subset of indexers are queryable from the a client application.
+Running multiple indexers that process inputs at the same rate are managed by a ``Coordinator``.
+The runtime view would look like:
+
+.. uml::
```

Comment:
Hmm, I think this part is missing from the webpage, at least for my local website. Could you double-check that it is appearing for you?

Reply:
It's working for me. I maybe forced push a fix without realizing that it was broken.

Reply:
Hmm, it's supposed to render a diagram, right? I still can't see it, but it must be something wrong with my settings